### ISSUE TITLE
Strange Yolo v11 recall results

### ISSUE THREAD

**Issue by Kaelrox** [ultralytics/ultralytics#22690]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

First,  I know there is a somewhat similar  issue with this post:

https://github.com/ultralytics/yolov5/issues/11194

But  as  is mentioned in the thread that is with v5 with different behaviours.

I have the following confusion matrix

<img width="811" height="542" alt="Image" src="https://github.com/user-attachments/assets/ca69f350-a714-4c98-a34a-43f2213302ba" />

Now consider my normalized confusion matrix

<img width="854" height="568" alt="Image" src="https://github.com/user-attachments/assets/cff9f7fb-b75f-42fa-8165-95b1d24704b9" />

Lastly my output 

<img width="871" height="215" alt="Image" src="https://github.com/user-attachments/assets/4dda5014-4269-45c9-9c04-eac0d7995dec" />

First a couple of confirmation questions:

1)  Is the bottom right always blank/0 because it is the combination of true and predicated background which does not occur because it would require  for a; the model to predict everything that is background and b; for the to be ground truths for everything that is background which would be chaos, correct?

2)  The  top right number in confusion matrix 1 is VERY high. Yet I have recall of 75%  and precision of 70% according to my output.  Why is this? Wouldn't the top right number completely ruin my recall and precision? Or is it that part of that top number is maybe parts of bounding boxes  that overlap too much into background? like my model is correctly identifying but drawing loose boxes?

Finally the main question:

Given my unnormalized confusion matrix, considering the values of 425 and 32, and my recall of 75%.  Recall = TP / (TP + FN).

Consider the following: 425/0.75 = 566. But, 425+32 = 457. What could account for this discrepancy?

Thanks !

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Kaelrox, thank you for your detailed question and for using Ultralytics 🚀! This is an automated response to help you get support as quickly as possible. An Ultralytics engineer will review your issue and follow up with you soon 🙂  

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since this looks like a custom training and metrics ❓ Question related to YOLO11 confusion matrices, precision, and recall, please:
- Provide as much information as possible about your training setup (model variant, training command, hyperparameters, image size, number of epochs, etc.)
- Include example validation images and the corresponding predictions if possible
- Share your full training and validation logs, especially the lines that show metrics computation
- Clarify exactly which plots come from which run/checkpoint (e.g. `confusion_matrix_normalized.png` from which epoch)

Also, please verify that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) ✅. This will help us understand whether the behavior you are seeing is expected or indicates an issue with the setup, the data, or the interpretation of the metrics.

If you believe this might actually be a 🐛 Bug (for example, a mismatch between reported metrics and the confusion matrix that you can reproduce consistently), then please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) that includes:
- A small subset of your dataset sufficient to reproduce the behavior
- The exact `yolo` command(s) or Python snippet you are running
- The version info from your environment (Python, `ultralytics`, PyTorch, CUDA)

This level of detail will greatly speed up troubleshooting and help the team give you an accurate explanation of your confusion matrix and metrics 📊.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Before diving deeper, please ensure you have upgraded to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/), to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
1. Yes.
2. The normalized value is across each column. Each column is normalized separately and sum to 1. The last column only has one cell, so it will be 1 regardless of the absolute value. It means "100% of your false positives were due to that class". See: https://github.com/ultralytics/ultralytics/issues/16695#issuecomment-2394795256

> Given my unnormalized confusion matrix, considering the values of 425 and 32, and my recall of 75%. Recall = TP / (TP + FN).
> 
> Consider the following: 425/0.75 = 566. But, 425+32 = 457. What could account for this discrepancy

The metrics in the logs are not based on the confusion matrix. See: https://github.com/ultralytics/ultralytics/issues/17047#issuecomment-2424660435

**Comment by odinmoja** [comment]
Hey, thanks so much for clearing that up for me. Appreciate you.

**Comment by glenn-jocher** [comment]
You're welcome, and glad it clarified things—if you have any more questions on YOLO11 metrics or evaluation, feel free to follow up here or open a new issue.
